sklearn/tests/test_common.py

```diff
@@ -222,7 +222,7 @@ def test_transformer_n_iter():
             else:
                 yield check_transformer_n_iter, name, estimator
 
-
+@ignore_warnings(category=DeprecationWarning)
```

Comment:
Hm, in the common tests we might allow that... Or maybe, can we inspect a class to see if it is deprecated? Below there is a hard-coded test for projected gradient. We should check that if the class is deprecated, we catch the deprecation warning, but not otherwise (it might call to deprecated functions, which we should fix).